x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -507,6 +508,73 @@ public void testBwcWithStoredAuthenticationHeaders() throws IOException {
         assertThat(authenticationV7.encode(), equalTo(headerV7));
     }
 
+    public void testMaybeRewriteForOlderVersionErasesDomainForVersionsBeforeDomains() {
+        final TransportVersion olderVersion = TransportVersionUtils.randomVersionBetween(
+            random(),
+            Authentication.VERSION_REALM_DOMAINS.calculateMinimumCompatVersion(),
```

Comment:
Looks like this is a deprecated API, and I think it's easy to find the replacement.
Also, hard-coded versions, especially for the bounds of the random generator, are OK too.

Reply:
Sounds good, I'll go with a hard-coded version -- unfortunately, there isn't a replacement method just a note stating "Code calling this should be refactored to not do this" 😅

Reply:
Ah nice, there's a new util method `randomPreviousCompatibleVersion` which does the trick.